src/classes/WalletConnectors/Metamask.ts

```diff
@@ -0,0 +1,17 @@
+import { BrowserConnector, type WindowProvider } from './BrowserConnector'
+
+export class MetaMaskWallet extends BrowserConnector {
+    public readonly id = 'metamask'
+    public readonly name: string = 'MetaMask'
+
+    public getProvider() {
+        if (typeof window === 'undefined') return
+
+        const { ethereum } = (window as unknown as { ethereum?: WindowProvider })
+
+        if (ethereum?.providers)
+            return ethereum.providers.find(_ethereum => _ethereum?.isMetaMask)
```

Comment:
at least for me, my metamask isn't injecting the `.providers` prop, is it injecting in another device?

Reply:
It depends wich wallet browser are you using. As I've stated above, some connectors can generate this array to avoid conflicts with a provider that is already injected on `window.ethereum`.

Take a look at this CoinBase documentation:
https://docs.cloud.coinbase.com/wallet-sdk/docs/injected-provider#multiple-injected-providers